lib/DDG/Meta/Data.pm

```diff
@@ -0,0 +1,170 @@
+package DDG::Meta::Data;
+# ABSTRACT: Metadata functions for instant answers
+
+use DDG::SpiceBundle::OpenSourceDuckDuckGo; 
+use DDG::GoodieBundle::OpenSourceDuckDuckGo; 
+use DDG::LongtailBundle::OpenSourceDuckDuckGo; 
+use DDG::FatheadBundle::OpenSourceDuckDuckGo; 
+use JSON::XS 'decode_json';
+use Path::Class;
+use File::ShareDir 'dist_file';
+use IO::All;
+use Clone 'clone';
+
+use strict;
+
+sub debug { 0 }
+use if debug, 'Data::Printer';
+
+no warnings 'uninitialized';
+
+# $ia_metadata => {
+#    id => { ... }
+#    module => { ... }
+#    sharedir => { ... }
+# }
+my %ia_metadata;
+
+my @ia_types = qw(Spice Goodie Longtail Fathead);
+
+my %metadata_files = map { $_ => dist_file("DDG-${_}Bundle-OpenSourceDuckDuckGo", lc $_ . '/meta/metadata.json') } @ia_types;
+
+# Only build metadata once. Not in BUILD so we can call apply_keywords directly
+unless(%ia_metadata){
+
+    FILE: while (my ($type, $filename) = each %metadata_files) {
+        warn "Processing IA type: $type with file $filename" if debug;
+
+        # One metadata file for each repo with the following format
+        # { "<IA name>": {
+        #     "id": " " 
+        #     "signal" : " "
+        #     ....
+        # }
+        my $file_data = decode_json(io($filename)->all);
+
+        # check for decode_json_file error, returns undef
+        die "reading metadata file failed ... $filename" unless $file_data;
+
+        IA: while (my ($id, $module_data) = each %{ $file_data }) {
+
+            # 20150502 (zt) Can't filter like this yet as some tests depend on non-live IA metadata
+            #next unless $module_data->{status} eq 'live';
+
+            # check for bad metadata.  We need a perl_module for the by_module key
+            if($module_data->{perl_module} !~ /DDG::.+::.+/){
+                warn "Something wrong with perl_module for IA $id perl_module: $module_data->{perl_module} in $filename ...  Not fatal but the IA won't show";
```

Comment:
Not a bad idea. I wanted to only load live IAs originally but we're not quite there yet.